dev_tools/requirements/re-pip-compile-in-docker.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+set -ex
+
+# We use docker to run the actual `pip-compile` command because its
+# behavior depends on the platform from which it is run. Please see
+# ./Dockerfile for the list of `pip-compile` commands that are run.
+docker build -t qualtran-pip-compile .
```

Comment:
I make pretty liberal use of 

```
# go/soverflow/59895/how-do-i-get-the-directory-where-a-bash-script-is-located
SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
readonly SCRIPT_DIR
```
In my build scripts so that I don't depend on the script being called from a specific directory. This SCRIPT_DIR can then inform the builds `.` and the cp's `./`

Reply:
done